Sources/CLI/BuildCommand.swift

```diff
@@ -102,6 +102,32 @@ extension Application {
         @Flag(name: .shortAndLong, help: "Suppress build output")
         var quiet: Bool = false
 
+        /// Checks for the existence of 'Containerfile' or 'Dockerfile' in the specified directory.
+        /// - Parameter buildDirectory: The directory to search within.
+        /// - Returns: The Path to the found 'Containerfile' or 'Dockerfile'.
+        /// - Throws: ValidationError if neither file is found.
+        ///
+        private static func checkForDockerOrContainerfile(in buildDirectory: String, using defaultBuildFileNames: Set<String>) throws -> Data {
```

Comment:
The function itself is generalized, and has nothing to do specifically with Dockerfile nor Containerfile. As you aim to return the path of the found build file, I will also refrain from using `check` as function verb and use `locate` instead.

```suggestion
        /// Locates the default build file (e.g. Dockerfile or Containerfile) in the specified directory.
        /// - Parameter buildDirectory: The directory to search within.
        /// - Returns: The Path to the found build file.
        /// - Throws: ValidationError if no default build file is found.
        ///
        private static func locateDefaultBuildFile(in buildDirectory: String, using defaultBuildFileNames: Set<String>) throws -> Data {
```